department-of-veterans-affairs/vets-api
# Added preferred name to users API

## Summary
Fetches the demographics and adds the preferred name to the profile. This also updates some tests that use the profile.

## Related issue(s)
https://app.zenhub.com/workspaces/mhv-on-vagov-landing-page-62619a987d74510018ecc546/issues/gh/department-of-veterans-affairs/va.gov-team/74398

## Testing done
- unit tests added and updated
- Manual testing using Demographics mock data (added via [this PR](https://github.com/department-of-veterans-affairs/vets-api/pull/15972) and [this PR](https://github.com/department-of-veterans-affairs/vets-api-mockdata/pull/463))
- See also [this PR](https://github.com/department-of-veterans-affairs/vets-api-mockdata/pull/464) for added mock data that can be used with mocked local authentication

Manual testing:
1. Checkout the `cfelix/74398-demographics-mock` branch for the `vets-api-mockdata` repo.
1. Checkout the branch for this PR for the `vets-api`
1. Start the `vets-api` locally
1. Start the `vets-website` locally
1. Access the [local mock login](http://localhost:3001/sign-in/mocked-auth)
1. Open the browser's developer tools and preserve the logs for the Network tab.
1. Use the mock login and login as an IDME user.
1. In the browser's developer tools network tab, find the call to the user's API (`user`) and verify that the preferred name is present. Note there is a default mock response for all users as well as a few specific responses in [this PR](https://github.com/department-of-veterans-affairs/vets-api-mockdata/pull/464).

## Screenshots
N/A

## What areas of the site does it impact?
- v0/user API

## Acceptance criteria


## Requested Feedback
This makes a new request when fetching the profile via the v0/user API. We added this call to be performed only when the user API gets a request and kept it out of the user model since it is not used anywhere else. Is there a preference here for this call to be in the user model, and if so, wouldn't that create a performance issue since user is used everywhere?
